File: AndroidApplication/app/src/main/java/com/crossp/client/AdData.java
package com.crossp.client;

public class AdData {
    public enum FileType { Image, Video }

    private Ad ad;
    public Ad getAd() { return ad; }

    public byte[] fileData;
    public byte[] getFileData() { return fileData; }

    private FileType type;
    public FileType getType() { return type; }

    {
        ad = null;
        fileData = null;
        type = null;
    }

    public AdData(Ad ad, byte[] fileData, FileType type)
    {
        this.ad = ad;
        this.fileData = fileData;
        this.type = type;
    }
}

File: UnityProject/Assets/Crossp/Scripts/Dtos/AdData.cs
namespace Crossp.DataTransferObjects
{
    public class AdData
    {
        public enum FileType { Image, Video }
        public AdDto Ad { get; private set; }
        public byte[] FileData { get; private set; }
        public FileType Type { get; private set; }

        public AdData(AdDto ad, byte[] fileData, FileType type)
        {
            Ad = ad;
            FileData = fileData;
            Type = type;
        }
    }
}
